service/internal/moduleinfo/moduleinfo.go

```diff
@@ -0,0 +1,20 @@
+// Copyright The OpenTelemetry Authors
+// SPDX-License-Identifier: Apache-2.0
+
+package moduleinfo // import "go.opentelemetry.io/collector/service/internal/moduleinfo"
+
+import "go.opentelemetry.io/collector/component"
+
+type ModuleInfo struct {
+	// BuilderRef is the raw string passed in the builder configuration used to build this service.
+	BuilderRef string
```

Comment:
Should we have version as well here in the future?

Reply:
Yes, I think it would make sense to move this to have more structure, but I would rather do this in a separate PR (I would have to start parsing the builder ref and I am worried that this is not trivial)